megamek/src/megamek/common/MULParser.java

```diff
@@ -1062,23 +1062,82 @@ private void setCrewAttributes(Crew crew, Map<String,String> attributes, Entity
      * @param attributes A map of attribute values keyed to the attribute names.
      */
     private void setPilotAttributes(Crew crew, int slot, Map<String, String> attributes) {
+        boolean hasGun = attributes.containsKey(GUNNERY) && (attributes.get(GUNNERY).length() > 0);
+        boolean hasRpgGun = attributes.containsKey(GUNNERYL) && (attributes.get(GUNNERYL).length() > 0) &&
+                attributes.containsKey(GUNNERYM) && (attributes.get(GUNNERYM).length() > 0) &&
+                attributes.containsKey(GUNNERYB) && (attributes.get(GUNNERYB).length() > 0);
+
         // Did we find required attributes?
-        if (!attributes.containsKey(GUNNERY) || (attributes.get(GUNNERY).length() == 0)) {
+        if (!hasGun && !hasRpgGun) {
             warning.append("Could not find gunnery for pilot.\n");
         } else if (!attributes.containsKey(PILOTING) || (attributes.get(PILOTING).length() == 0)) {
             warning.append("Could not find piloting for pilot.\n");
         } else {
             // Try to get a good gunnery value.
             int gunVal = -1;
-            try {
-                gunVal = Integer.parseInt(attributes.get(GUNNERY));
-            } catch (NumberFormatException ignored) {
-                // Handled by the next if test.
+            if (hasGun) {
+                try {
+                    gunVal = Integer.parseInt(attributes.get(GUNNERY));
+                } catch (NumberFormatException ignored) {
+                    // Handled by the next if test.
+                }
+
+                if ((gunVal < 0)
+                        || (gunVal > Crew.MAX_SKILL)) {
+                    warning.append("Found invalid gunnery value: ")
+                            .append(attributes.get(GUNNERY)).append(".\n");
+                    return;
+                }
             }
-            if ((gunVal < 0) || (gunVal > Crew.MAX_SKILL)) {
-                warning.append("Found invalid gunnery value: ")
-                        .append(attributes.get(GUNNERY)).append(".\n");
-                return;
+
+            // get RPG skills
+            int gunneryLVal = gunVal;
+            int gunneryMVal = gunVal;
+            int gunneryBVal = gunVal;
+            if (hasRpgGun) {
+                if ((attributes.containsKey(GUNNERYL)) && (attributes.get(GUNNERYL).length() > 0)) {
+                    try {
+                        gunneryLVal = Integer.parseInt(attributes.get(GUNNERYL));
+                    } catch (NumberFormatException ignored) {
+                        // Handled by the next if test.
```

Comment:
If they have a GUNNERY attribute that is valid, when this falls thru you won't receive any helpful message. You'll just silently use the value from that attribute.

Reply:
Updated